Code/BuildSystem/CMake/ezUtils.cmake

```diff
@@ -188,9 +196,11 @@ function(ez_set_common_target_definitions TARGET_NAME)
 	target_compile_definitions(${TARGET_NAME} PRIVATE BUILDSYSTEM_SDKVERSION_MINOR="${EZ_CMAKE_SDKVERSION_MINOR}")
 	target_compile_definitions(${TARGET_NAME} PRIVATE BUILDSYSTEM_SDKVERSION_PATCH="${EZ_CMAKE_SDKVERSION_PATCH}")
 
+	set(ORIGINAL_BUILD_TYPE "$<IF:$<STREQUAL:${EZ_CMAKE_GENERATOR_CONFIGURATION},${EZ_BUILDTYPENAME_DEBUG}>,DEBUG,$<IF:$<STREQUAL:${EZ_CMAKE_GENERATOR_CONFIGURATION},${EZ_BUILDTYPENAME_DEV}>,DEV,SHIPPING>>")
+
 	# set the BUILDSYSTEM_BUILDTYPE definition
-	target_compile_definitions(${TARGET_NAME} PRIVATE BUILDSYSTEM_BUILDTYPE="${EZ_CMAKE_GENERATOR_CONFIGURATION}")
-	target_compile_definitions(${TARGET_NAME} PRIVATE BUILDSYSTEM_BUILDTYPE_${EZ_CMAKE_GENERATOR_CONFIGURATION})
+	target_compile_definitions(${TARGET_NAME} PRIVATE "BUILDSYSTEM_BUILDTYPE=\"${ORIGINAL_BUILD_TYPE}\"")
+	target_compile_definitions(${TARGET_NAME} PUBLIC "BUILDSYSTEM_BUILDTYPE_${ORIGINAL_BUILD_TYPE}")
```

Comment:
Why was this change necessary ?

Reply:
We have a `#ifdef BUILDSYSTEM_BUILDTYPE_Shipping` in UserConfig.h
If you do not use Debug|Dev|Shipping but the CMake default Debug|RelWithDebInfo|Release instead, then the this `#ifdef` would never trigger.

Furthermore because various `#ifdef BUILDSYSTEM_BUILDTYPE_XXX` statements are visible from public headers, the visibility has to be changed from PRIVATE to PUBLIC otherwise they are only defined when building Foundation and not compiling something that consumes foundation.